src/Middleware/OutputCaching/src/OutputCacheKeyProvider.cs

```diff
@@ -119,6 +159,12 @@ public void AppendVaryByKey(OutputCacheContext context, StringBuilder builder)
             {
                 var header = varyByRules.HeaderNames[i] ?? string.Empty;
                 var headerValues = requestHeaders[header];
+
+                if (ContainsDelimiters(header))
```

Comment:
Header names cannot contain control characters either.

Reply:
I was able to set it by code, but since we also use an allow-list for these I removed the check.